mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go

```diff
@@ -530,6 +551,9 @@ func resourceBigtableInstanceClusterReorderTypeList(_ context.Context, diff *sch
 	for i := 0; i < newCount.(int); i++ {
 		_, newId := diff.GetChange(fmt.Sprintf("cluster.%d.cluster_id", i))
 		_, c := diff.GetChange(fmt.Sprintf("cluster.%d", i))
+		if _, ok := clusters[newId.(string)]; ok {
```

Comment:
I think it would be better to move this to its own CustomizeDiff function, and add that to `customdiff.All()`. This function is concerned specifically with the ordering of clusters between runs, and so it makes some assumptions (like not running on create). It is a little bit confusing to have the unique id validation live inside here.

On that same note, if you use a second CustomizeDiff function that runs for both create and update, I think you could safely remove the `verifyDuplicatedClusters()` checks (unless I'm missing something).

Reply:
Good idea! Didn't know I could do that :) Done!